cumulusci/tasks/bulkdata/mapping_parser.py

```diff
@@ -86,6 +100,12 @@ class MappingStep(CCIDictModel):
     ] = None  # default should come from task options
     anchor_date: Optional[Union[str, date]] = None
 
+    @validator("bulk_mode", "api", "action", pre=True)
+    def case_normalize(cls, val):
+        if isinstance(val, Enum):
+            return val
+        return ENUM_VALUES.get(val.lower())
```

Comment:
I tested with an invalid value, and the error is pretty inscrutable:

```
2021-03-31 17:12:10: Getting org info from Salesforce CLI for test-fnuu4th1ktiu@example.com
2021-03-31 17:12:14: Beginning task: GenerateAndLoadDataFromYaml
2021-03-31 17:12:14: As user: test-fnuu4th1ktiu@example.com
2021-03-31 17:12:14: In org: 00D21000000JHHa
2021-03-31 17:12:14: 
2021-03-31 17:12:14: Beginning task: GenerateDataFromYaml
2021-03-31 17:12:14: 

1 validation error for SObjectRuleDeclarationFile
__root__ -> 0 -> api
  unexpected value; permitted: 'smart', 'rest', 'bulk' (type=value_error.const; given=bulko; permitted=('smart', 'rest', 'bulk'))
```

Reply:
I'll make a WI to improve this in the future as we discussed.